### DISCUSSION TITLE
Welcome to foundryvtt-fallout Discussions!

### DISCUSSION THREAD

**Discussion by Muttley** [Muttley/foundryvtt-fallout#519] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by QckSilverDragon** [comment]
Just want to say that i really appreciate everything you have done to contribute to Fallout2D20 in Foundry.

I have been a hobbyist become "professional" developer of C# windows applications for years. I have dabbled in many languages over the years so javascript isnt new. Web development is new.

After seeing your wiki post a few weeks ago about how to contribute i decided to give it a go. Ill try and follow your conventions while i learn html and web development. I like to jump into the deep end so im starting by working on the vehicle sheet that i requested some time ago.

Heres to hoping something i do gets pulled in. 🍻 👍

**Reply by Muttley** [reply]
Great stuff.  I look forward to seeing what you come up with.

**Comment by MorganBard** [comment]
Hello all! Is there a way to add a Source to the source list? I've added some homebrew content and would like to source it as such, merely for the sake of completeness.

Also, as a veteran C# developer, is there any way to develop modules in C#? Frankly, I dread working in Javascript. It is so completely different from languages in which I've worked.

Many thanks!

**Reply by QckSilverDragon** [reply]
You can make a custom module with compendiums that can be recognized as a source. Muttley has made an example.
https://github.com/Muttley/foundryvtt-fallout-custom-module

As for development, i dont know if you can use pure C#. You may have to ask in the dev channel in the Foundry discord server.
 If you look at the source for the Fallout2d20 system you can see how similar this is to C# already. I am a primary C# dev but am able to pick up JS pretty easily due to the similarity.

**Reply by MorganBard** [reply]
It's similar in syntax, that's about where the similarity ends (at least IMO). Object orientation doesn't work the same way, inheritance doesn't seem to work the same way. Anyways, I've tried. I've used it extensively in client-side web development, but never for something like an actual application. It just seems... clunky, I guess is the best word to describe it.

**Comment by MorganBard** [comment]
So, here's a more difficult question. One of the things we've found with the Fallout 2D20 game system as written by Modiphius is a severe lack of skills covering social functions and other areas. I've added a separate compendium with a skill list as I and my players like it. How do I get new characters to be based on that compendium instead of the built-in one? I've added skills to the built-in one, after unlocking it, but, of course, those will be wiped out when the system updates. Is there a relatively painless way to do what I need?

Incidentally, this is a duplicate of the post on the Discord. Apologies if that's inappropriate. Let me know and I shall restrict myself to one venue or the other.

Thanks again :)

Morgan

**Reply by Muttley** [reply]
Add them to your own compendiums and they won't get overwritten.  For maximum usability add them to your own custom module that you can then load into any Fallout world you create.

There's an example custom module with compendiums built in you can use here: https://github.com/Muttley/foundryvtt-fallout-custom-module

Obviously make sure you backup your custom compendiums regularly.

**Reply by QckSilverDragon** [reply]
I have custom skills in my custom module and they dont get added to new character sheets. If i add the skills to the core compendium, they get added.

**Reply by Muttley** [reply]
> I have custom skills in my custom module and they dont get added to new character sheets. If i add the skills to the core compendium, they get added.

Only the core Skills are added automatically as of right now.

**Reply by QckSilverDragon** [reply]
Thanks for the clarification. i haven't looked into new document creation process yet.
May be worth including them as part of the source selection setting.

**Comment by DiddyDragon** [comment]
Hi, maybe I'm doing something wrong but wanted to ask and see. I'm setting up and testing stuff in preperation for running some games and came across the Scavenging Locations stuff. I set up the different roll tables for loot and set up the default RollTable settings but when I press 'Scavenge Location' only Caps, Pre-War money and Junk actually show up in Found Items. Is there something else that I'm supposed to do to make it roll for loot on the tables?

**Reply by Muttley** [reply]
> Hi, maybe I'm doing something wrong but wanted to ask and see. I'm setting up and testing stuff in preperation for running some games and came across the Scavenging Locations stuff. I set up the different roll tables for loot and set up the default RollTable settings but when I press 'Scavenge Location' only Caps, Pre-War money and Junk actually show up in Found Items. Is there something else that I'm supposed to do to make it roll for loot on the tables?

What are the min/max settings for the scavenging site you created?  Those would determine how many rolls should be done on the tables.

**Reply by DiddyDragon** [reply]
I've tried it with a few different settings, even setting the minimum to 1 on every table, and tried putting points into the AP Spend boxes and using the pre-set min/max options. Nothing rolls, only Junk, caps and Pre-war money (images are my last attempt and the result)
![image](https://github.com/user-attachments/assets/968eefee-919c-4da5-9ec5-ee8c8bbf5353)
![image](https://github.com/user-attachments/assets/156c5414-d076-4577-9c2d-34bcf797c94f)

**Reply by QckSilverDragon** [reply]
Whats one of the roll tables look like?
Are all the results in the roll tables items or just text?
You must have them be actual inventory items. 
I think they also need to be in a compendium. Not 100% on that.

**Reply by DiddyDragon** [reply]
The tables are all items except for the "other" table that is a mix of item and text. They're in a custom compendium
![image](https://github.com/user-attachments/assets/4c180b02-5017-4486-9e1e-1c8db7aa90c3)

**Reply by Muttley** [reply]
> The tables are all items except for the "other" table that is a mix of item and text. They're in a custom compendium

That's probably the issue, I suspect there are errors in the console.  You can *only* use item (or other rollable tables) entries as it's too complex to parse text to work out what to add.

**Reply by DiddyDragon** [reply]
Changed the other table to one with just items, still the same result, and no errors in the console at any point. All that shows up in the console when I hit "Scavenge Location" is "ChatTimestamp.addGameTimeToMessage      index.js:1"

**Reply by QckSilverDragon** [reply]
I believe the roll table and the roll result must both be a compendium item. 
![image](https://github.com/user-attachments/assets/8ab73771-8f84-4524-ac9f-1252acd6440e)

**Reply by DiddyDragon** [reply]
That works! Thank you so much for all the help

**Comment by Nevets1701** [comment]
With the new way that mods are added to weapons, how can I make it so custom weapons can receive mods?

**Reply by Muttley** [reply]
Just drag/drop the mods you want the weapon to have access to onto the weapon's item sheet.

**Reply by Nevets1701** [reply]
Thank you sir

**Comment by NateJackson99** [comment]
Is there anything in the works, or any plans to add the content from all of the other books and/or erratas?

**Reply by QckSilverDragon** [reply]
Modiphius has only allowed the player facing content from the CRB. You would have to petition them for some premium style content.

**Comment by Stevie4711** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:

Hey! 
1) was the Foundry channel for Fallout (and other game systems) cancelled?
2) If I deleted a skill (e.g. repair) - could you help me how I get it back? :-D

Many thanks and again, thanks for the great work!!

**Reply by Muttley** [reply]
> > ## 👋 Welcome!
> > We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> Hey!
> 
>     1. was the Foundry channel for Fallout (and other game systems) cancelled?
> 
>     2. If I deleted a skill (e.g. repair) - could you help me how I get it back? :-D
> 
> 
> Many thanks and again, thanks for the great work!!

1.  It's right here where it;s always been: https://discord.com/channels/170995199584108546/753726151985332266
2. Drag it back onto the sheet from the system compendiums.
